ncm-systemd/src/main/perl/Systemd/Systemctl.pm

```diff
@@ -311,27 +311,32 @@ Returns exitcode and output.
 sub systemctl_command_units
 {
     my ($logger, $command, @units) = @_;
+    my ($success, $failure);
 
-    # TODO: any relevant options?
-    my $proc = CAF::Process->new(
-        [$SYSTEMCTL, $command],
-        log => $logger,
-        );
+    $success = $failure = 0;
 
-    if (@units) {
-        $proc->pushargs('--', @units);
+    # TODO: any relevant options?
```

Comment:
is there any reference where this behaviour is defined? also, CAF::Service works like this, so might need changes as well

Reply:
No, `man systemctl` is quiet about what happens if you specify multiple units and one or more causes a failure. The biggest problem is the lack of indication _which_ unit caused the failure. So yes, I think CAF::Service should also handle one unit at a time.